src/sentry/runner/initializer.py

```diff
@@ -539,3 +543,101 @@ def on_configure(config):
 
     if 'south' in settings.INSTALLED_APPS:
         skip_migration_if_applied(settings, 'social_auth', 'social_auth_association')
+
+
+def validate_snuba():
+    """
+    Make sure everything related to Snuba is in sync.
+
+    This covers a few cases:
+
+    * When you have features related to Snuba, you must also
+      have snuba fully configured correctly to continue.
+    * If you have Snuba specific search/tagstore/tsdb backends,
+      you must also have a snuba compatible eventstream backend
+      otherwise no data getting written into Snuba.
+    * If you only have Snuba related eventstream, yell that you
+      probably want the other backends otherwise things are weird.
+    """
+    if not settings.DEBUG:
+        return
+
+    from sentry.features import requires_snuba as snuba_features
+
+    snuba_enabled_features = set()
+
+    for feature in snuba_features:
+        if settings.SENTRY_FEATURES.get(feature, False):
+            snuba_enabled_features.add(feature)
+
+    snuba_required_features = bool(snuba_enabled_features)
+
+    # TODO: TSDB is left out since... just becacuse for now.
+    snuba_required_backends = (
+        settings.SENTRY_SEARCH == 'sentry.search.snuba.SnubaSearchBackend' or
+        settings.SENTRY_TAGSTORE == 'sentry.tagstore.snuba.SnubaCompatibilityTagStorage'
+    )
+
+    eventstream_is_snuba = (
+        settings.SENTRY_EVENTSTREAM == 'sentry.eventstream.snuba.SnubaEventStream' or
+        settings.SENTRY_EVENTSTREAM == 'sentry.eventstream.kafka.KafkaEventStream'
+    )
+
+    # All good here, it doesn't matter what else is going on
+    if snuba_required_backends and eventstream_is_snuba:
+        return
+
+    if snuba_required_features and not (eventstream_is_snuba or snuba_required_backends):
+        from .importer import ConfigurationError
+        show_big_error('''
+You have features enabled which require Snuba, but you don't have Snuba.
```

Comment:
"have Snuba" here is confusing, more clarification around what they don't have could be helpful.